spec/takeaway_spec.rb

```diff
@@ -4,15 +4,17 @@
   let(:menu) { double :menu }
   let(:dish) { double :dish }
   let(:order) { double :order }
+  let(:text_message) { double :text_message }
   let(:order_class) { double :order_class, new: order }
-  subject(:takeaway) { Takeaway.new(menu, order_class) }
+  subject(:takeaway) { Takeaway.new(menu, order_class, text_message) }
 
-  it "should return a menu object when view_menu method is called" do
+  it 'should return a menu object when view_menu method is called' do
     expect(takeaway.view_menu).to eq menu
   end
 
-  it "should return order when place_order method is called succesfully" do
+  it 'should send confirmation text when place_order method is called succesfully' do
     allow(order).to receive(:create_order) { order }
-    expect(takeaway.place_order({ dish: dish, number: 2 }, 10)).to eq order
+    allow(text_message).to receive(:send_text) { 'text sent' }
```

Comment:
You are stubbing out the behaviour of the text_message for your RSpec tests which is good to see :)